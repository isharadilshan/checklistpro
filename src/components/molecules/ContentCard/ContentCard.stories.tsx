import React from 'react';
import {NativeBaseProvider} from 'native-base';
import {ComponentMeta, ComponentStory} from '@storybook/react';

import ContentCard from './';

export default {
  title: 'components/molecules/ContentCard',
  component: ContentCard,
} as ComponentMeta<typeof ContentCard>;

export const Basic: ComponentStory<typeof ContentCard> = () => {
  const args = {
    title: 'STORYBOOK-CARD',
    imgUrl: 'https://picsum.photos/seed/picsum/200/300',
    onPress: () => {},
  };
  return (
    <NativeBaseProvider>
      <ContentCard {...args} />
    </NativeBaseProvider>
  );
};
