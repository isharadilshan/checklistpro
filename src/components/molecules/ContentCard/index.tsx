import React from 'react';
import {TouchableOpacity} from 'react-native';
import {Box, AspectRatio, Image, Center} from 'native-base';
import createStyle from './styles';

type CardItem = {
  title: string;
  imgUrl: string;
  onPress: () => void;
};

const ContentCard: React.FC<CardItem> = ({title, imgUrl, onPress}) => {
  const styles = createStyle();

  return (
    <TouchableOpacity onPress={onPress}>
      <Box maxW="40" style={styles.cardWrapper}>
        <Box>
          <AspectRatio w="100%" ratio={4 / 3}>
            <Image
              source={{
                uri: imgUrl,
              }}
              alt="image"
            />
          </AspectRatio>
          <Center
            style={styles.taglineWrapper}
            _text={{
              color: 'warmGray.50',
              fontWeight: '700',
              fontSize: 'xs',
            }}
          >
            {title}
          </Center>
        </Box>
      </Box>
    </TouchableOpacity>
  );
};

export default ContentCard;
