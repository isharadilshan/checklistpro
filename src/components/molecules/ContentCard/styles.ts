import {ViewStyle, StyleSheet} from 'react-native';

interface Style {
  cardWrapper: ViewStyle;
  taglineWrapper: ViewStyle;
}

export default () => {
  return StyleSheet.create<Style>({
    cardWrapper: {
      borderWidth: 1,
      borderRadius: 4,
      margin: 10,
      overflow: 'hidden',
      borderColor: '#e5e7eb',
      backgroundColor: '#e5e7eb',
    },
    taglineWrapper: {
      bottom: 0,
      paddingVertical: 2,
      paddingHorizontal: 6,
      position: 'absolute',
      backgroundColor: '#7c3aed',
    },
  });
};
