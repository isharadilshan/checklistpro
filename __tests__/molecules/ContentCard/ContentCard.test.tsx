import React from 'react';
import {describe, expect} from '@jest/globals';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import ContentCard from '../../../src/components/molecules/ContentCard';

configure({adapter: new Adapter()});

const props = {
  title: 'STORYBOOK-CARD',
  imgUrl: 'https://picsum.photos/seed/picsum/200/300',
  onPress: jest.fn(),
};

describe('when rendering', () => {
  const wrapper = shallow(<ContentCard {...props} />);

  it('should render a component matching the snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});
