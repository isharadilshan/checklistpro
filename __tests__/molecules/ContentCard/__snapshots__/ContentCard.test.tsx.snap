// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering should render a component matching the snapshot 1`] = `
<TouchableOpacity
  onPress={[MockFunction]}
>
  <Memo(ForwardRef(Box))
    maxW="40"
    style={
      {
        "backgroundColor": "#e5e7eb",
        "borderColor": "#e5e7eb",
        "borderRadius": 4,
        "borderWidth": 1,
        "margin": 10,
        "overflow": "hidden",
      }
    }
  >
    <Memo(ForwardRef(Box))>
      <Memo(ForwardRef(AspectRatio))
        ratio={1.3333333333333333}
        w="100%"
      >
        <Memo(ForwardRef)
          alt="image"
          source={
            {
              "uri": "https://picsum.photos/seed/picsum/200/300",
            }
          }
        />
      </Memo(ForwardRef(AspectRatio))>
      <Memo(ForwardRef(Center))
        _text={
          {
            "color": "warmGray.50",
            "fontSize": "xs",
            "fontWeight": "700",
          }
        }
        style={
          {
            "backgroundColor": "#7c3aed",
            "bottom": 0,
            "paddingHorizontal": 6,
            "paddingVertical": 2,
            "position": "absolute",
          }
        }
      >
        STORYBOOK-CARD
      </Memo(ForwardRef(Center))>
    </Memo(ForwardRef(Box))>
  </Memo(ForwardRef(Box))>
</TouchableOpacity>
`;
